enhancements/coreos-encrypted-by-default.md

```diff
@@ -0,0 +1,422 @@
+
+Rather than provide a new-new module, the upstream module would be modified to support Clevis configurations and submitted to the Clevis upstream.
+
+### CoreOS Installer
+
+The current [CoreOS Installer](https://github.com/coreos/coreos-installer/blob/master/coreos-installer) is a Dracut module. At a high-level it:
+* parses the boot options
+* downloads the image
+* writes it disk
+* saves parameters into /boot
+* reboots
+
+This design does not require any changes to the Installer. Once the CoreOS installer is done, the on-disk CoreOS will be booted and re-encrypted normally.
+
+### Risks and Mitigations
+
+#### Caveat Emptor
+The clearest risk to the design is that while the data will be encrypted by default, the installation will determine **if a passphrase is in plain text** (this makes it trivial for an attacker to decrypt the data).  There are only two cases where the keys are held secure:
+* the user provides a Clevis configuration that uses a secure key escrow (Tang or KMS)
+* the hardware or Cloud supports TPM2
+
+This risk cannot be understated and should be clearly and explicitly called out in the user documentation. There is an axiom in encryption that no encryption is better than bad encryption as it can create a false sense of security.
+
+With LUKS, [a passphrase is *NOT* the master key used for encrpytion](https://gitlab.com/cryptsetup/cryptsetup/blob/master/FAQ#L56-63); the master key is encrypted in the LUKS header on disk. Passphrases are used to unlock the master key. Each passphrase is stored in a "key-slot," which can be removed (`cryptsetup luksKillSlot...`). [An attacker could use an old snapshot or block-level backup to defeat encryption](https://gitlab.com/cryptsetup/cryptsetup/blob/master/FAQ#L56-63). If a user cycles a Clevis configuration, then existing block-level backups and snapshots should be secured unless a re-encryption operation is performed.
+
+##### LUKS overhead: null cipher
+
+When using the null cipher (no encryption), the overhead is negligible. The "overhead" really becomes cosmetic and introduces more steps to open a null-encrypted container.
+
+To illistrate the point, the current disk format mount process is simply: `mount /dev/disk/by-label/rootfs /mnt`, while the new process would be:
+```
+# clevis-luks-unlock \
+      -d /dev/disk/by-label/coreos_cryptfs \
+      coreos_cryptfs
+
+# mount /dev/mapper/coreos_cryptfs /mnt
+```
+Note: the above examples assume a Linux rescue terminal with the dependencies.
+
+#### Drift between FCOS and RHCOS
+
+There is a considerable question as to the desirability of this feature in Fedora CoreOS. Both Red Hat CoreOS and Fedora CoreOS have different goals: the former is targeted at Openshift clusters, while the later is targeted at running container workloads (including clusters). There is some concern that the Fedora CoreOS community will not find this solution desirable and thus Red Hat will use a LUKS container while Fedora will not.
+
+### Test Plan
```

Comment:
This is a big one. This is likely something we'll want to be happening in CI I'd think and verified there.

Reply:
:100: to this point. I wanted to this out for initial approval since the matrix is going to be rather in-depth.